engine/classes/Elgg/Cache/EntityCache.php

```diff
@@ -89,7 +89,7 @@ public function getByUsername($username) {
 	public function set(ElggEntity $entity) {
 		$guid = $entity->guid;
 
-		if (!$guid || isset($this->entities[$guid]) || isset($this->disabled_guids[$guid])) {
```

Comment:
I don't know why this was here, but it's wrong. We propagate changes after `ElggEntity::update()`